js/README.md

```diff
@@ -100,55 +100,163 @@ Subdirectories with additional utilities
-`basicblocks.js` and `logo.js`.
+<!-- * You can add an individual block to a palette by modifying two files:
+`basicblocks.js` and `logo.js`. -->
+* To add a new block you first need to look through `js/blocks/` and determine if a new file is needed or not. If required then go through the section down below on how to declare new files for blocks requiring so.
 
-* If the block you are adding needs to expand into a stack of blocks,
-you may also need to modify `macro.js`.
+## How to define new files for blocks
 
-* If you want to add a new palette with multiple blocks for a specific
+Note: You may skip this section if the block you're trying to add doesn't require a new file.
+
+* Make a new file in `js/blocks` with a meaningful name.
+
+e.g. Current files are named as  `GraphicsBlocks.js` , `MediaBlocks.js`.
+
+* Add that file to `MUSICBLOCKS_EXTRAS` in `js/activity.js`.
+
+* Create a setup function in your new file at the end, with a meaningful name.
+
+e.g. `setupGraphicsBlocks()`.
+
+* Call that setup function in `js/basicblocks.js` inside `initBasicProtoBlocks()` function.
+
+<!-- * If the block you are adding needs to expand into a stack of blocks,
+you may also need to modify `macro.js`. -->
+
+<!-- * If you want to add a new palette with multiple blocks for a specific
 application, you may want to write a plugin. Please see:
 [plugin](http://github.com/sugarlabs/musicblocks/tree/master/plugins)
-instead.
+instead. -->
 
-## How to define a new block in basicblocks.js (deprecated)
+## How to define a new block
 
+Note: You should directly start with this step if the block you want to add doesn't require a new file and makes use of an existing file inside `js/blocks`.
+<!-- 
 Note: New blocks are now added to the appropriate file in the `blocks`
 subdirectory. Much of the discussion below is still somewhat relevant
-as background reading.
+as background reading. -->
 
 [basicblocks.js](https://github.com/sugarlabs/musicblocks/blob/master/js/basicblocks.js)
-is where each block is defined, its palette assigned, its shape and
-label defined, and any default arguments assigned.
+is the file where setup function related to each block file is called.
 
-[logo.js](https://github.com/sugarlabs/musicblocks/blob/master/js/logo.js)
-is where the code associated with running each block is defined.
+<!-- [logo.js](https://github.com/sugarlabs/musicblocks/blob/master/js/logo.js)
+is where the code associated with running each block is defined. -->
 
-1. Define the type of block
+1. Create a new class inside the file. All demonstrations extend from `BaseBlock`, however blocks can extend each other also.
 
-`var uniquenameBlock = new ProtoBlock('uniquename');`
+e.g. `class uniqueNameBlock extends AnotherBlock{}`
 
-e.g., `var pitchNumberBlock = new ProtoBlock('pitchnumber');`
+* As part of MVC structure the argument earlier passed to `new ProtoBlock()` is now passed to `super()` call inside constructor of the defined class.
```

Comment:
Why mention the old way of doing things at all?

Reply:
I'll restructure that. I thought it'll be helpful for users migrating from the previous version.